ddtrace/profiling/_threading.pyx

```diff
@@ -23,42 +21,34 @@ cpdef get_thread_name(thread_id):
         # we fail, it'll just be an anonymous thread because it's either
         # starting or dying.
         try:
-            return threading_mod._active[thread_id].name
+            return threading_mod._active[thread_id]
         except KeyError:
             try:
-                return threading_mod._limbo[thread_id].name
+                return threading_mod._limbo[thread_id]
             except KeyError:
                 pass
 
     return None
 
 
-cpdef get_thread_native_id(thread_id):
-    # Do not force-load the threading module if it's not already loaded
-    if "threading" not in sys.modules:
-        return None
+cpdef get_thread_name(thread_id):
+    thread = get_thread_by_id(thread_id)
+    return thread.name if thread is not None else None
 
-    import threading
 
-    try:
-        thread_obj = threading._active[thread_id]
-    except KeyError:
-        try:
-            thread_obj = ddtrace_threading._active[thread_id]
-        except KeyError:
-            # This should not happen, unless somebody started a thread without
-            # using the `threading` module.
-            # In that case, well… just use the thread_id as native_id 🤞
-            return thread_id
+cpdef get_thread_native_id(thread_id):
+    thread = get_thread_by_id(thread_id)
+    if thread is None:
+        return thread_id
 
     try:
         # We prioritize using native ids since we expect them to be surely unique for a program. This is less true
         # for hashes since they are relative to the memory address which can easily be the same across different
         # objects.
-        return thread_obj.native_id
+        return thread.native_id
     except AttributeError:
         # Python < 3.8
-        return hash(thread_obj)
+        return hash(thread)
```

Comment:
`return hash(thread)` wasn't introduced in this PR, but I wonder what the consequences of trying to do better might be.

`hash(thread)` will return a hash that isn't guaranteed to be stable even when the underlying native thread + python thread are the same, right?  The caller is asking specifically for a quantity that is less volatile than `thread_id`, and we're giving it something more volatile.

We might choose to call `gettid()` (https://man7.org/linux/man-pages/man2/gettid.2.html), but
* We'd need to figure out the corresponding call on Windows
* we add a syscall to every call of `get_thread_native_id()` (and it isn't clear to me how to cache these)


I don't think this needs to be fixed in this PR, but I'm curious to hear your thoughts.  For my part, I vaguely feel like we're doing the wrong thing (although I acknowledge this mostly affects older Pythons, so it isn't critical).

Reply:
You are totally right on all your points. The problem with `gettid` is that it would always be called from the main thread, which would then normally return the process ID 🙁 . There is a similar problem for macOS and that's why we don't have true CPU time there. A possible (but perhaps not very clean) way of working around this would be to wrap the `Thread` class to call and store `gettid()` as first thing after the bootstrap, as done in #2849). Another hacky way would be to _guess_ where in the `pthread_t` structure the `tid` field is, and read the value from there (cf. https://github.com/P403n1x87/austin/blob/b48d980f6f3d0808d9221ee611a9c3094e02bb50/src/linux/py_proc.h#L690-L734)

Reply:
Thank you for the thoughts and explanations!  Sounds like you've thought this through :)